controllers/secret.go

```diff
@@ -72,7 +94,7 @@ func (r *OperatorPipelineReconciler) reconcilePyxisAPISecret(meta metav1.ObjectM
 		Name:      PYXIS_API_SECRET,
 	}
 
-	secret := newSecret(key)
+	secret := newSecret(key, "", "")
```

Comment:
I am curious to know why we don't create an instance of `metav1.ObjectMeta` instead of `types.NamespacedName` to pass to the newSecret method? In this case we don't need to do conversion in newSecret method.

Reply:
Since this method is used by all three secret types, the `NamespacedName.Name` needs to be different for each struct. The `meta` struct does not have this info. If we go with your approach, the `newSecret` method would then have four params, correct?

note: you commented on the pyxis method which isn't implemented, just needed `""` passed in bc of the refactor.

Reply:
Can't we modify newSecret to receive metav1.ObjectMeta, instead of `types.NamespacedName`?

Reply:
Per our call, the `types.NamespacesName` is required for `client.Get(context.TODO(), key, obj)` call in the controller-runtime.